src/chat/chat.service.ts

```diff
@@ -0,0 +1,40 @@
+import { BaseMessage } from 'bee-agent-framework/llms/primitives/message';
+import { LLMError } from 'bee-agent-framework/llms/base';
+
+import {
+  ChatCompletionCreateBody,
+  ChatCompletionCreateResponse
+} from './dtos/chat-completion-create';
+
+import { createChatLLM, getDefaultModel } from '@/runs/execution/factory';
+import { getLogger } from '@/logger';
+import { APIError, APIErrorCode } from '@/errors/error.entity';
+import { generatePrefixedObjectId } from '@/utils/id';
+
+const getChatLogger = () => getLogger();
+
+export async function createChatCompletion({
+  model = getDefaultModel(),
+  messages
+}: ChatCompletionCreateBody): Promise<ChatCompletionCreateResponse> {
+  const llm = createChatLLM({ model });
+  try {
+    const output = await llm.generate(
+      messages.map((message) => new BaseMessage(message.role, message.content))
```

Comment:
Shouldn't we use some template based on the model?

Reply:
I don't think we can even do that in general, the llm backend does that for us if it supports chat endpoint.

@Tomas2D what do you think?